device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java

```diff
@@ -129,17 +130,15 @@ public void open(Queue<DeviceClientConfig> deviceClientConfigs, ScheduledExecuto
                 {
                     this.log.trace("MQTT connection will use sas token based auth");
                     this.iotHubUserPassword = this.config.getSasTokenAuthentication().getRenewedSasToken(false, false);
+                    this.webSocketQueryString = NO_CLIENT_CERT_QUERY_STRING;
                 }
                 else if (this.config.getAuthenticationType() == DeviceClientConfig.AuthType.X509_CERTIFICATE)
                 {
-                    if (this.config.isUseWebsocket())
-                    {
-                        //Codes_SRS_MQTTIOTHUBCONNECTION_34_027: [If this function is called while using websockets and x509 authentication, an UnsupportedOperationException shall be thrown.]
-                        throw new UnsupportedOperationException("X509 authentication is not supported over MQTT_WS");
-                    }
-
                     this.log.trace("MQTT connection will use X509 certificate based auth");
 
+                    this.webSocketQueryString = "";
```

Comment:
So this needs to be set to empty string? Or should there be conditional check when appending this?

Reply:
Sure, I can leave it null for SAS cases, and just check when building the string later